test/jdk/java/awt/Frame/InitialMaximizedTest/InitialMaximizedTest.java

```diff
@@ -21,197 +21,82 @@
-  to read and understand someone else's test.
- ****************************************************/
-
-/**
- This is part of the standard test machinery.
- It creates a dialog (with the instructions), and is the interface
-  for sending text messages to the user.
- To print the instructions, send an array of strings to Sysout.createDialog
-  WithInstructions method.  Put one line of instructions per array entry.
- To display a message for the tester to see, simply call Sysout.println
-  with the string to be displayed.
- This mimics System.out.println but works within the test harness as well
-  as standalone.
- */
-
-class Sysout
-{
-    private static TestDialog dialog;
-
-    public static void createDialogWithInstructions( String[] instructions )
-    {
-        dialog = new TestDialog( new Frame(), "Instructions" );
-        dialog.printInstructions( instructions );
-        dialog.setVisible(true);
-        println( "Any messages for the tester will display here." );
+public class InitialMaximizedTest {
+
+    static Frame frame;
+
+    public static void main(String[] args) throws Exception {
+        if (!Toolkit.getDefaultToolkit()
+                    .isFrameStateSupported(Frame.MAXIMIZED_BOTH)) {
+            return;
+        }
+
+        Robot robot = new Robot();
+        try {
+            EventQueue.invokeAndWait(InitialMaximizedTest::createAndShowFrame);
+            robot.waitForIdle();
+            robot.delay(2000);
+            EventQueue.invokeAndWait(InitialMaximizedTest::checkMaximized);
+        } finally {
+            EventQueue.invokeAndWait(() -> {
+                if (frame != null) {
+                    frame.dispose();
+                }
+            });
+        }
     }
 
-    public static void createDialog( )
-    {
-        dialog = new TestDialog( new Frame(), "Instructions" );
-        String[] defInstr = { "Instructions will appear here. ", "" } ;
-        dialog.printInstructions( defInstr );
-        dialog.setVisible(true);
-        println( "Any messages for the tester will display here." );
-    }
+    private static void checkMaximized() {
+        Rectangle frameBounds = frame.getBounds();
 
+        GraphicsConfiguration gc = frame.getGraphicsConfiguration();
+        Rectangle workArea = gc.getBounds();
 
-    public static void printInstructions( String[] instructions )
-    {
-        dialog.printInstructions( instructions );
-    }
+        Insets screenInsets = Toolkit.getDefaultToolkit().getScreenInsets(gc);
+        workArea.x += screenInsets.left;
+        workArea.y += screenInsets.top;
+        workArea.width -= screenInsets.left + screenInsets.right;
+        workArea.height -= screenInsets.top + screenInsets.bottom;
 
+        System.out.println("Frame bounds " + frameBounds);
+        System.out.println("GraphicsConfiguration bounds " + gc.getBounds());
+        System.out.println("Screen insets: " + screenInsets);
+        System.out.println("Work area: " + workArea);
```

Comment:
Is it intentional to keep so many print statement?

Reply:
Looks cleaner to me than
```java
System.out.println(
        "Frame bounds " + frameBounds +
        "\nGraphicsConfiguration bounds " + gc.getBounds() +
        "\nScreen insets: " + screenInsets +
        "\nWork area: " + workArea
);
```